crates/mpt/src/errors.rs

```diff
@@ -0,0 +1,64 @@
+//! Errors for the `kona-derive` crate.
+
+use alloc::string::String;
+use thiserror::Error;
+
+/// A [Result] type alias where the error is [TrieNodeError].
+pub type TrieNodeResult<T> = Result<T, TrieNodeError>;
+
+/// An error type for [TrieNode] operations.
+///
+/// [TrieNode]: crate::TrieNode
+#[derive(Error, Debug, PartialEq, Eq)]
+pub enum TrieNodeError {
+    /// Invalid trie node type encountered.
+    #[error("Invalid trie node type encountered")]
+    InvalidNodeType,
+    /// Failed to decode trie node.
+    #[error("Failed to decode trie node: {0}")]
+    RLPError(alloy_rlp::Error),
+    /// Key does not exist in trie.
+    #[error("Key does not exist in trie. Encountered {0} node.")]
+    KeyNotFound(String),
+    /// Trie node is not a leaf node.
+    #[error("Trie provider error: {0}")]
+    Provider(String),
+}
+
+/// A [Result] type alias where the error is [TrieDBError].
+pub type TrieDBResult<T> = Result<T, TrieDBError>;
```

Comment:
It seems off to me that we've removed the `DB` abbreviation in other types like `TrieProvider` and `TrieHinter` but use it here. Would it not make sense to call this `TrieResult` as well? Would favor consistency if possible, otherwise it's difficult for grugbrain.dev to understand.